app/src/main/java/net/sourceforge/opencamera/ExtendedAppInterface.java

```diff
@@ -205,31 +295,469 @@ public void onFrameInfoRecordingFailed() {
+
+            manualSeekbars = extendedAppInterface.mMainActivity.getManualSeekbars();
+        }
+
+        @Override
+        protected Void doInBackground(SyncSettingsContainer... settingsContainers) {
+            Log.d(TAG, "doInBackground");
+
+            if (settingsContainers.length != 1) {
+                throw new IllegalArgumentException(
+                        String.format("One SyncSettingsContainer must be passed but was %d", settingsContainers.length)
+                );
+            }
+            final SyncSettingsContainer settings = settingsContainers[0];
+
+            final ExtendedAppInterface extendedAppInterface = extendedAppInterfaceRef.get();
+            final MainActivity mainActivity = mainActivityRef.get();
+            final Preview preview = previewRef.get();
+            final MainUI mainUI = mainUIRef.get();
+            if (extendedAppInterface == null || mainActivity == null || preview == null || mainUI == null) {
+                return null;
+            }
+
+            waitForCameraToOpen(); // If camera is not opened wait until it is
+
+            // Close some UI elements for the changes to be reflected in them
+            mainActivity.runOnUiThread(() -> {
+                mainUI.closeExposureUI();
+                mainUI.closePopup();
+                mainUI.destroyPopup();
+            });
+
+            syncCaptureFormat(settings);
+
+            cameraController = preview.getCameraController();
+
+            syncExposure(settings);
+            if (settings.syncISO) syncISO(settings);
+            if (settings.syncWb) syncWb(settings);
+            if (settings.syncFlash) syncFlash(settings);
+            if (settings.syncFormat) syncFormat(settings);
+
+            extendedAppInterface.getDrawPreview().updateSettings(); // Ensure that the changes get cached
+            return null;
+        }
+
+        private void waitForCameraToOpen() {
+            Preview preview = previewRef.get();
+            if (preview == null) return;
+
+            Log.d(TAG, "Waiting for camera to open...");
+
+            while (!preview.openCameraAttempted()) {
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+        }
+
+        private void syncCaptureFormat(SyncSettingsContainer settings) {
+            MainActivity mainActivity = mainActivityRef.get();
+            Preview preview = previewRef.get();
+            if (mainActivity == null || preview == null) return;
+
+            Log.d(TAG, "Syncing settings: capture format");
+
+            if (preview.isVideo() != settings.isVideo) {
+                mainActivity.runOnUiThread(() -> mainActivity.clickedSwitchVideo(null));
+
+                waitForCameraToOpen(); // Need to wait until camera gets reopened
+
+                // Need to wait for capture mode to change
```

Comment:
We could put this busy waiting in `FutureTask `and use `get() `method to wait for result